src/core/Akka/Util/Internal/SynchronizationContextManager.cs

```diff
@@ -0,0 +1,72 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ArrayExtensions.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Akka.Annotations;
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Akka.Util.Internal
+{
+    /// <summary>
+    /// SynchronizationContextManager controls SynchronizationContext of the async pipeline.
+    /// <para>Does the same thing as .ConfigureAwait(false) but better - it should be written once only,
+    /// unlike .ConfigureAwait(false).</para>
+    /// <para> await SynchronizationContextManager.RemoveContext; 
+    /// Should be used as a very first line inside async public API of the library code</para>
+    /// </summary>
+    /// <example> 
+    /// This sample shows how to use <see cref="SynchronizationContextManager"/> .
+    /// <code>
+    /// class CoolLib 
+    /// {
+    ///     public async Task DoSomething() 
+    ///     {
+    ///         await SynchronizationContextManager.RemoveContext;
+    ///         
+    ///         await DoSomethingElse();
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    [InternalApi]
+    internal static class SynchronizationContextManager
+    {
+        public static ContextRemover RemoveContext { get; } = new ContextRemover();
+    }
+
+    [InternalApi]
+    internal class ContextRemover : INotifyCompletion
+    {
+        public bool IsCompleted => SynchronizationContext.Current == null;
+
+        public void OnCompleted(Action continuation)
+        {
+            var prevContext = SynchronizationContext.Current;
+
+            try
+            {
+                SynchronizationContext.SetSynchronizationContext(null);
+                continuation();
+            }
+            finally
+            {
+                SynchronizationContext.SetSynchronizationContext(prevContext);
+            }
+        }
+
+        public ContextRemover GetAwaiter()
+        {
+            return this;
+        }
+
+        public void GetResult()
```

Comment:
Is there meant to be something here?

Reply:
Nah... this method there is to make this class awaitable.